scipy/special/_basic.py

```diff
@@ -3074,36 +3113,65 @@ def factorialk(n, k, exact=True):
     Parameters
     ----------
     n : int or array_like
-        Calculate multifactorial. If `n` < 0, the return value is 0.
+        Calculate multifactorial. If ``n < 0``, the return value is 0.
     k : int
         Order of multifactorial.
     exact : bool, optional
         If exact is set to True, calculate the answer exactly using
-        integer arithmetic.
+        integer arithmetic, otherwise use an approximation (faster,
+        but yields floats instead of integers)
+
+        .. warning::
+           The default value for ``exact`` will be changed to
+           ``False`` in SciPy 1.15.0.
 
     Returns
     -------
     val : int
         Multifactorial of `n`.
 
-    Raises
-    ------
-    NotImplementedError
-        Raises when exact is False
-
     Examples
     --------
     >>> from scipy.special import factorialk
-    >>> factorialk(5, 1, exact=True)
+    >>> factorialk(5, k=1, exact=True)
     120
-    >>> factorialk(5, 3, exact=True)
+    >>> factorialk(5, k=3, exact=True)
     10
+    >>> factorialk([5, 7, 9], k=3, exact=True)
+    array([ 10,  28, 162])
+    >>> factorialk([5, 7, 9], k=3, exact=False)
+    array([ 10.,  28., 162.])
+
+    Notes
+    -----
+    While less straight-forward than for the double-factorial, it's possible to
+    calculate a general approximation formula of n!(k) by studying ``n`` for a given
+    remainder ``r < k`` (thus ``n = m * k + r``, resp. ``r = n % k``), which can be
+    put together into something valid for all integer values ``n >= 0`` & ``k > 0``::
+
+      n!(k) = k ** ((n - r)/k) * gamma(n/k + 1) / gamma(r/k + 1) * max(r, 1)
 
+    This is the basis of the approximation when ``exact=False``. Compare also [1].
+
+    Additionally, the factorialk function ensures that 0-dimensional arrays as inputs
+    produce the same type upon return (as opposed to silent conversion to a scalar).
```

Comment:
I think we should mention what the default behavior is, and the motivation behind breaking the convention for combinatorial functions. Maybe saomething like

The default behavior for [ufuncs](https://numpy.org/doc/stable/reference/ufuncs.html) is to silently convert the output to scalar for 0-dimensional array input but `factorialk` and other combinatorial functions in SciPy will preserve shape when given  0-dimensional input. This is because in some cases they will return object arrays of arbitrary sized integers, and conversion to scalar will result in the output being a Python ``int`` object which would be missing the array attributes and methods which are present for NumPy scalars.

Reply:
I'm not sure I'd want to call it "default behaviour" as much as "inherited legacy behaviour". There's no good a priori reason for converting the type, and in the RFC there seemed to be broad agreement that this is more historical artefact rather than intentional.

I'll reformulate the note as you wish, but I feel this is getting lost too much in the weeds to be useful for a docstring TBH.

Reply:
I’d say don’t reformulate it if you don’t agree. I meant it as a suggestion, and wouldn’t want you to make the change without consensus. At the moment, I think the one sentence seems lacking in context, but maybe the ideal thing would be to link somewhere else that has more detailed information, leaving the details out of the docstring. Honestly; I think it’s fine to punt on documenting this behavior for now and leave mention of this behavior for a future PR. That way we can avoid holding this up without forcing through something suboptimal.

Reply:
OK, I removed the last commit for now, it seems we should also deal with this question independently from this PR. :)